core/src/main/java/org/elasticsearch/index/analysis/AnalysisService.java

```diff
@@ -262,10 +261,13 @@ public AnalysisService(Index index, @IndexSettings Settings indexSettings, @Null
             }
         }
 
-        defaultAnalyzer = analyzers.get("default");
+        NamedAnalyzer defaultAnalyzer = analyzers.get("default");
         if (defaultAnalyzer == null) {
             throw new IllegalArgumentException("no default analyzer configured");
         }
+        if (analyzers.containsKey("default_index") && Version.indexCreated(indexSettings).onOrAfter(Version.V_2_0_0)) {
```

Comment:
I had planned to completely remove it for master, should I keep the error instead?